### PR TITLE
[nav_hybrid] max acceleration and deceleration

### PR DESCRIPTION
**Pull request by dewagter**

### CODE REVIEW

File: sw/airborne/modules/nav/nav_rotorcraft_hybrid.c [hunk 1]
```diff
@@ -227,7 +261,20 @@ static void nav_hybrid_circle(struct EnuCoor_f *wp_center, float radius)
...
   float_vect2_normalize(&speed_unit);
-  VECT2_SMUL(nav.speed, speed_unit, desired_speed);
+
+  struct FloatVect2 speed_sp;
+
+  VECT2_SMUL(speed_sp, speed_unit, desired_speed);
+
+  // Check for acceleration bound
+  float abs_nav_speed = sqrtf(nav.speed.x * nav.speed.x + nav.speed.y * nav.speed.y);
+  float abs_speed_sp = sqrtf(speed_sp.x * speed_sp.x + speed_sp.y * speed_sp.y);
```
Comment (gautierhattenberger) [inline sw/airborne/modules/nav/nav_rotorcraft_hybrid.c:271]: use `float_vect2_norm` for readability ?

File: sw/airborne/modules/nav/nav_rotorcraft_hybrid.c [hunk 2]
```diff
@@ -126,7 +144,23 @@ static void nav_hybrid_route(struct EnuCoor_f *wp_start, struct EnuCoor_f *wp_en
...
   float length_direction = Max(float_vect2_norm(&direction), 0.01f);
   // Scale to have the desired speed
-  VECT2_SMUL(nav.speed, direction, desired_speed / length_direction);
+  struct FloatVect2 speed_sp;
+  
+  VECT2_SMUL(speed_sp, direction, desired_speed / length_direction);
+
+  // Check for acceleration bound
+  float abs_nav_speed = sqrtf(nav.speed.x * nav.speed.x + nav.speed.y * nav.speed.y);
+  float abs_speed_sp = sqrtf(speed_sp.x * speed_sp.x + speed_sp.y * speed_sp.y);
```
Comment (gautierhattenberger) [inline sw/airborne/modules/nav/nav_rotorcraft_hybrid.c:153]: use `float_vect2_norm` or `FLOAT_VECT2_NORM` for readability ?

File: sw/airborne/modules/nav/nav_rotorcraft_hybrid.c [hunk 3]
```diff
@@ -76,13 +78,21 @@ static void nav_hybrid_goto(struct EnuCoor_f *wp)
...
-    float max_speed_decel2 = fabsf(2.f * dist_to_wp * MAX_DECELERATION); // dist_to_wp can only be positive, but just in case
+    float max_speed_decel2 = fabsf(2.f * dist_to_wp * nav_max_deceleration_sp); // dist_to_wp can only be positive, but just in case
     float max_speed_decel = sqrtf(max_speed_decel2);
     // Bound the setpoint velocity vector
     float max_h_speed = Min(nav_max_speed, max_speed_decel);
     float_vect2_bound_in_2d(&speed_sp, max_h_speed);
   }
 
+  // Check for acceleration bound
+  float abs_nav_speed = sqrtf(nav.speed.x * nav.speed.x + nav.speed.y * nav.speed.y);
```
Comment (gautierhattenberger) [inline sw/airborne/modules/nav/nav_rotorcraft_hybrid.c:89]: the macro `FLOAT_VECT2_NORM` can be used here

File: sw/airborne/modules/nav/nav_rotorcraft_hybrid.c [hunk 4]
```diff
@@ -34,6 +34,8 @@
 #endif
 #define NAV_MAX_SPEED (GUIDANCE_INDI_MAX_AIRSPEED + GUIDANCE_INDI_NAV_SPEED_MARGIN)
 float nav_max_speed = NAV_MAX_SPEED;
+float nav_max_acceleration_sp = 6.0;
+float nav_max_deceleration_sp = 0.5;
 
 #ifndef MAX_DECELERATION
 #define MAX_DECELERATION 1.f
```
Comment (gautierhattenberger) [inline sw/airborne/modules/nav/nav_rotorcraft_hybrid.c:41]: max acceleration and deceleration should be made configurable

**Review by gautierhattenberger — CHANGES_REQUESTED** [review]

File: sw/airborne/modules/nav/nav_rotorcraft_hybrid.c (hunk 4 above)
Comment (dewagter) [reply sw/airborne/modules/nav/nav_rotorcraft_hybrid.c]: ok

File: sw/airborne/modules/nav/nav_rotorcraft_hybrid.c [hunk 5]
```diff
@@ -35,10 +35,13 @@
 #define NAV_MAX_SPEED (GUIDANCE_INDI_MAX_AIRSPEED + GUIDANCE_INDI_NAV_SPEED_MARGIN)
 float nav_max_speed = NAV_MAX_SPEED;
 
-#ifndef MAX_DECELERATION
-#define MAX_DECELERATION 1.f
+#ifndef NAV_HYBRID_MAX_DECELERATION
+#define NAV_HYBRID_MAX_DECELERATION 2.0
 #endif
 
+float nav_max_acceleration_sp = NAV_HYBRID_MAX_ACCELERATION;
```
Comment (dewagter) [inline sw/airborne/modules/nav/nav_rotorcraft_hybrid.c:42]:
```suggestion
```

**Review by EwoudSmeur — APPROVED** [review]
Looks good, thanks!

**Pull request merged**